doc/contributing/docs/sphinx_warnings.rst

```diff
@@ -0,0 +1,194 @@
+Sphinx-build warnings reference
+-------------------------------
+
+This document will guide you through the possible warnings raised by Sphinx engine
+while building the docs.
+
+Warnings and solutions list
+===========================
```

Comment:
H2 should use `-`

Reply:
And I doubt that this document needs third-level headings at all. Can we make them all h2?

Reply:
I think H2 is too big. I used H2 specifically to reach H3